src/main/java/org/openhab/tools/analysis/checkstyle/PomXmlCheck.java

```diff
@@ -139,20 +137,29 @@ private void processManifestFile(File file) throws CheckstyleException {
         manifestBundleSymbolicName = bundleInfo.getSymbolicName();
     }
 
-    private void processPomXmlFile(File file, String[] lines) throws CheckstyleException {
+    private void processPomXmlFile(FileText fileText) throws CheckstyleException {
+        File file = fileText.getFile();
         File pomDirectory = file.getParentFile();
+
         // the pom directory path will be used in the finalization
         pomDirectoryPath = pomDirectory.getPath();
         File parentPom = new File(pomDirectory.getParentFile(), POM_XML_FILE_NAME);
 
         // The pom.xml must reference the correct parent pom (which is usually in the parent folder)
         if (parentPom.exists()) {
-            String parentArtifactIdValue = getNodeValue(file, POM_PARENT_ARTIFACT_ID_XPATH_EXPRESSION);
-            String parentPomArtifactIdValue = getNodeValue(parentPom, POM_ARTIFACT_ID_XPATH_EXPRESSION);
+            FileText parentPomFileText = null;
+            try {
+                parentPomFileText = new FileText(parentPom, "UTF-8");
+            } catch (IOException e) {
+                logger.error("Error in reading the pom file", e);

```

Comment:
But can and should you proceed what will happen on line 158 when passing in a null?

Reply:
This seems unanswered?

Reply:
I added a return on line 155. The code no longer proceeds if exception is caught.

Reply:
Sorry, was having breakfast with the kids, so only checked for comment to be outdated 😊